lib/metric-config-parser/metric_config_parser/metric.py

```diff
@@ -247,16 +251,31 @@ def resolve(
                 ]
                 metric_definition.statistics = self.statistics
                 metric_summary = metric_definition.resolve(spec, conf, configs)
+                metric_definition.aggregation_units = self.aggregation_units or [
+                    AnalysisUnit.CLIENT
+                ]
```

Comment:
What's the purpose of this `elif metric_definition` block here? A metric definition is missing a select expression and a data source, so we pull the rest of the definition from the config collection and overwrite its aggregation units? I think I'm a bit lost.

Reply:
Yea the idea of this code is that if the current metric is missing a `select_expression` or `data_source`, it is probably augmenting/overriding a metric defined somewhere else. We get that definition with `metric_definition = configs.get_metric_definition(self.name, conf.app_name)` and then override a few fields.

What I'm not clear on is why we only override a few select fields here and not just take every field that is defined. @scholtzan any thoughts here?

Reply:
Yeah, this is to override a metric defined somewhere else. I think these few select fields are the ones relevant for analysis. Might make sense to also override the other available fields

Reply:
Should we gate this on `self.aggregation_units is not None`? What if the metric elsewhere (`metric_definition`) explicitly specifies `profile_group_id` as the `aggregation_unit` and the overriding metric does not specify anything? In that case, this will override the profile unit with a client unit, right? Is that intended?

Reply:
Yea good call. I added a test for this to verify the behavior you described, and fixed it by adding another option to the `or` chain. Now we'll use (in order) the `self` value if available, or the `metric_definition` if available, or the default if neither is defined.

Of note, this was the same for `analysis_bases` so fixed that here too.